packages/ember-metal/lib/computed.js

```diff
@@ -382,49 +393,65 @@ ComputedPropertyPrototype.get = function(obj, keyName) {
   @return {Object} The return value of the function backing the CP.
   @public
 */
-ComputedPropertyPrototype.set = function computedPropertySetWithSuspend(obj, keyName, value) {
-  var oldSuspended = this._suspended;
+ComputedPropertyPrototype.set = function computedPropertySetEntry(obj, keyName, value) {
+  if (this._readOnly) {
+    throw new EmberError(`Cannot set read-only property "${keyName}" on object: ${inspect(obj)}`);
+  }
 
-  this._suspended = obj;
+  if (!this._setter) {
+    return this.clobberSet(obj, keyName, value);
```

Comment:
I remember that at some point in the past we considered to warn/deprecate in this case, unless you mark the CP like overridable, because it's usually not what you intend to to. 

Just checking if it's a good moment to do that.